### ISSUE TITLE
Publishing fails on rerun.

### ISSUE THREAD

**Issue by mmitche** [dotnet/arcade#4922]
-  Is this issue blocking no
-  Is this issue causing unreasonable pain yes

When the publishing stage fails, rerunning it does not necessarily result in the desired result, which is that publishing should be able to effectively continue onwards, checking that each already uploaded item is identical to the one we want to publish. Instead, publishing crashes.

https://dev.azure.com/dnceng/internal/_build/results?buildId=536364&view=logs&j=7d245b7a-7558-514d-6701-fbe8f979cd93&t=00d86622-4ce2-524c-e924-3574a1000a6d

**Comment by JohnTortugo** [comment]
The failure reported in the linked build, in specific, has a fix in newer versions of the SDK: https://github.com/dotnet/arcade/pull/4867

I believe "retry" currently works. I'm going to confirm that.

**Comment by JohnTortugo** [comment]
I validated the behavior using this build: https://dnceng.visualstudio.com/internal/_build/results?buildId=538436&view=results steps:

- I added a step to the publishing job that checks for existence of a blob file in Azure. 
- In the first execution attempt of the job the blob didn't exist and the job failed as expected. 
- Then I created the blob and re-started the job this time the publishing job succeeded without problem.
